Core/Scripting:Fix gossips not load from db

leelf00 opened the pull request:
<!--- (**********************************)
      (** Fill in the following fields **)
      (**********************************) --->

**Changes proposed:**

Fix gossip option not load from db.
- [x] Blackrock Depths
- [x] Karazhan
- [x] Battle for Mount Hyjal
- [x] Zulfarrak
- [x] Trail of the Champion 
- [x] Halls of Stone  
- [x] Quest The Hunt is On (ID 11794)
- [x] Quest Last Rites (ID 12109)
- [x] Magisters Terrace
- [x] Escape from DurnholdeKeep
- [x] Serpent Shrine Cavern

  
updates https://github.com/TrinityCore/TrinityCore/issues/22696

**Tests performed:**

build and tested with zhTW client in game.


<!--- Notes
- Enable the setting "[√] Allow edits from maintainers." when creating your pull request.
- If this PR only contains SQL files, open a new issue instead and post or link the SQL in the issue.
- When adding new SQL files, name them 9999_99_99_99_db_name.sql to reduce the chance of possible merge conflicts.
--->

leelf00 committed:
fix gossip option not load from db

fix blackrock depths,karazhan,hyjal,zulfarrak,trail of the champion halls of stone    gossip option not load from db.

MaxtorCoder commented:
Please rename the sql file to `9999_99_99_world.sql`, the person who will merge this PR will rename it correctly.

leelf00 committed:
rename sql update filename

leelf00 commented:
updated filename

leelf00 committed:
updae caps

leelf00 committed:
update capz

leelf00 commented on sql/updates/world/3.3.5/9999_99_99_world.sql:
```diff
@@ -0,0 +1,15 @@
+-- Battle Of Mount Hyjal missing optionid
+DELETE FROM gossip_menu_option where MenuID=7552 and OptionID in (0,1,2);
+DELETE FROM gossip_menu_option where MenuID=7581 and OptionID in (0,1,2);
+DELETE FROM gossip_menu_option where MenuID=7706 and OptionID=0;
```
updated

leelf00 committed:
change Quest The Hunt is On (ID 11794)  gossips load from db

change Quest The Hunt is On(11794)  gossips load from db

Aokromes commented on sql/updates/world/3.3.5/9999_99_99_world.sql:
```diff
@@ -0,0 +1,15 @@
+DELETE FROM gossip_menu_option WHERE MenuID=7552 AND OptionID IN (0,1,2);
+DELETE FROM gossip_menu_option WHERE MenuID=7581 AND OptionID IN (0,1,2);
+DELETE FROM gossip_menu_option WHERE MenuID=7706 AND OptionID=0;
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES (7552, 0, 0, 'My companions and I are with you, Lady Proudmoore.', 14698, 0, 0, 7556, 0, 0, 0, NULL, 0, 26365),(7552, 1, 0, 'We are ready for whatev ...
```
add new row after VALUES

Aokromes commented on sql/updates/world/3.3.5/9999_99_99_world.sql:
```diff
@@ -0,0 +1,15 @@
+
+-- HallsOfStone missing optionid
+DELETE FROM gossip_menu_option WHERE MenuID=9670 AND OptionID=0;
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES (9670, 0, 0, 'Let\'s move Brann, enough of the history lesson!', 27616, 0, 0, 0, 0, 0, 0, NULL, 0, 0);
```
missing newline.

leelf00 committed:
add new line

Aokromes commented on sql/updates/world/3.3.5/9999_99_99_world.sql:
```diff
@@ -0,0 +1,24 @@
+DELETE FROM gossip_menu_option WHERE MenuID=7552 AND OptionID IN (0,1,2);
+DELETE FROM gossip_menu_option WHERE MenuID=7581 AND OptionID IN (0,1,2);
+DELETE FROM gossip_menu_option WHERE MenuID=7706 AND OptionID=0;
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES 
```
remove ghost whitespace after VALUES :)

leelf00 committed:
remove whitespace

Aokromes commented on sql/updates/world/3.3.5/9999_99_99_world.sql:
```diff
@@ -14,11 +14,11 @@ INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText
 DELETE FROM gossip_menu_option WHERE MenuID=9670 AND OptionID=0;
-INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES 
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES
 (9670, 0, 0, 'Let\'s move Brann, enough of the history lesson!', 27616, 0, 0, 0, 0, 0, 0, NULL, 0, 0);
```
still missing newline :)

leelf00 replied:
add newline before insert?

Aokromes replied:
no at end of file.

leelf00 committed:
add newline

leelf00 committed:
Merge branch '3.3.5' into fix_battleformounthyjal_zulfarrak_blackrockdepths_karazhan_trialofchampion_locale

leelf00 committed:
move more text to DB

move Magisters Terrace,Escape from DurnholdeKeep,Serpent Shrine Cavern,Quest Last Rites (ID 12109) text to DB.

leelf00 committed:
remove whitespace

Aokromes commented:
it looks good to me.

offl committed:
adjustments

leelf00 commented:
thank for adjuest，but Magisters Terrace       code come back to #define.

offl commented:
That script used non-existing gossips

Aokromes commented:
updated https://github.com/TrinityCore/TrinityCore/issues/22696

leelf00 commented:
Magisters Terrace Gossips seems belong to npc Kalecgos (https://tbc.wowhead.com/npc=24848/kalecgos#comments).I check menuid exists in gossip_menu_option,any other thing missing?
